### PR TITLE
adding logs if ntp is out of sync for [OPS-1395]

### PR DESCRIPTION
**Pull request by bilalahmad99**
@edx/devops plz review

### CODE REVIEW

File: playbooks/roles/ecommerce/templates/edx/app/ecommerce/log-ntp-alerts.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+reach=$(ntpq -c associations | awk '{print $5}' | grep yes)
+if [[ ${reach} == *"no"* ]]; then
+    echo "NTPD not synchronized - Please investigate" >> ${log_directory}/ntp.log
+fi
+
+limit=100   # limit in milliseconds
+offsets=$(ntpq -nc peers | tail -n +3 | cut -c 62-66 | tr -d '-')
+for offset in ${offsets}; do
+    if [ ${offset:-0} -ge ${limit:-100} ]; then
+        echo "An NTPD offset is excessive - Please investigate" >> ${log_directory}/ntp.log
```
Comment (feanil) [inline playbooks/roles/ecommerce/templates/edx/app/ecommerce/log-ntp-alerts.sh.j2]: Please add a timestamp as well.

**Comment by feanil** [comment]
Code generally looks good but I think it belongs it common since I think we want to know if this is the case with any IDA.

File: playbooks/roles/ecommerce/templates/edx/app/ecommerce/log-ntp-alerts.sh.j2 (hunk 1 above)
Comment (bilalahmad99) [inline playbooks/roles/ecommerce/templates/edx/app/ecommerce/log-ntp-alerts.sh.j2]: done

File: playbooks/roles/common/templates/log-ntp-alerts.sh.j2 [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+reach=$(ntpq -c associations | awk '{print $5}' | grep yes)
+if [[ ${reach} == *"no"* ]]; then
+    echo $(date -u) "NTPD not synchronized - Please investigate" >> ${log_directory}/ntp.log
+fi
+
+limit=100   # limit in milliseconds
+offsets=$(ntpq -nc peers | tail -n +3 | cut -c 62-66 | tr -d '-')
+for offset in ${offsets}; do
+    if [ ${offset:-0} -ge ${limit:-100} ]; then
+        echo $(date -u) "An NTPD offset is excessive - Please investigate" >> ${log_directory}/ntp.log
```
Comment (maxrothman) [inline playbooks/roles/common/templates/log-ntp-alerts.sh.j2]: Please also log the amount of offset and the hostname of the server (also above)

Comment (bilalahmad99) [inline playbooks/roles/common/templates/log-ntp-alerts.sh.j2]: done

Comment (feanil) [inline playbooks/roles/common/templates/log-ntp-alerts.sh.j2]: Add a space after the date time stamp.  `" An`

@maxrothman  the hostname will be available via splunk.  +1 to putting the offset in here

**Comment by feanil** [comment]
:+1:   Don't forget to add an item to our list of splunk log items in secure config.

File: playbooks/roles/common/templates/log-ntp-alerts.sh.j2 (hunk 2 above)
Comment (maxrothman) [inline playbooks/roles/common/templates/log-ntp-alerts.sh.j2]: Good point @feanil: hostname can be left out.

**Pull request merged**